### ISSUE TITLE
ChannelProxy referenced from a method is not visible from class loader

### ISSUE THREAD

**Issue by amitoj** [spring-projects/spring-amqp#1008]
Why am I suddenly getting this error after moving Spring Boot app from Windows to Linux?

Windows JDK - 
java version "11.0.1" 2018-10-16 LTS
Java(TM) SE Runtime Environment 18.9 (build 11.0.1+13-LTS)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.1+13-LTS, mixed mode)

Linux JDK (attempt 1) -
openjdk version "11.0.3" 2019-04-16 LTS
OpenJDK Runtime Environment 18.9 (build 11.0.3+7-LTS)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.3+7-LTS, mixed mode, sharing)

Linux JDK (attempt 2) -
java version "11.0.3" 2019-04-16 LTS
Java(TM) SE Runtime Environment 18.9 (build 11.0.3+12-LTS)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.3+12-LTS, mixed mode)

RabbitMQ 3.7.14 running on Docker same host as Spring Boot app

```
java.lang.IllegalArgumentException: org.springframework.amqp.rabbit.connection.ChannelProxy referenced from a method is not visible from class loader
        at java.base/java.lang.reflect.Proxy$ProxyBuilder.ensureVisible(Proxy.java:858) ~[na:na]
        at java.base/java.lang.reflect.Proxy$ProxyBuilder.validateProxyInterfaces(Proxy.java:681) ~[na:na]
        at java.base/java.lang.reflect.Proxy$ProxyBuilder.<init>(Proxy.java:627) ~[na:na]
        at java.base/java.lang.reflect.Proxy$ProxyBuilder.<init>(Proxy.java:635) ~[na:na]
        at java.base/java.lang.reflect.Proxy.lambda$getProxyConstructor$0(Proxy.java:415) ~[na:na]
        at java.base/jdk.internal.loader.AbstractClassLoaderValue$Memoizer.get(AbstractClassLoaderValue.java:329) ~[na:na]
        at java.base/jdk.internal.loader.AbstractClassLoaderValue.computeIfAbsent(AbstractClassLoaderValue.java:205) ~[na:na]
        at java.base/java.lang.reflect.Proxy.getProxyConstructor(Proxy.java:413) ~[na:na]
        at java.base/java.lang.reflect.Proxy.newProxyInstance(Proxy.java:1006) ~[na:na]
        at org.springframework.amqp.rabbit.connection.CachingConnectionFactory.getCachedChannelProxy(CachingConnectionFactory.java:619) ~[spring-rabbit-2.1.5.RE
        at org.springframework.amqp.rabbit.connection.CachingConnectionFactory.getChannel(CachingConnectionFactory.java:498) ~[spring-rabbit-2.1.5.RELEASE.jar!/
        at org.springframework.amqp.rabbit.connection.CachingConnectionFactory.access$1600(CachingConnectionFactory.java:101) ~[spring-rabbit-2.1.5.RELEASE.jar!
        at org.springframework.amqp.rabbit.connection.CachingConnectionFactory$ChannelCachingConnectionProxy.createChannel(CachingConnectionFactory.java:1331) ~
        at org.springframework.amqp.rabbit.core.RabbitTemplate.doExecute(RabbitTemplate.java:2079) ~[spring-rabbit-2.1.5.RELEASE.jar!/:2.1.5.RELEASE]
        at org.springframework.amqp.rabbit.core.RabbitTemplate.lambda$execute$13(RabbitTemplate.java:2036) ~[spring-rabbit-2.1.5.RELEASE.jar!/:2.1.5.RELEASE]
        at org.springframework.retry.support.RetryTemplate.doExecute(RetryTemplate.java:287) ~[spring-retry-1.2.4.RELEASE.jar!/:na]
        at org.springframework.retry.support.RetryTemplate.execute(RetryTemplate.java:180) ~[spring-retry-1.2.4.RELEASE.jar!/:na]

```

**Comment by artembilan** [comment]
Try to update to the latest Spring Boot.
Looks like what you need has been fixed in Spring AMQP 2.1.6: https://github.com/spring-projects/spring-amqp/issues/967

**Comment by amitoj** [comment]
Thanks, upgrading Spring Boot version fixed the issue.
